app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -507,6 +507,10 @@ public function saveEntity($entity, $unlock = true)
             $details = $ips->first()->getIpDetails();
             // Only update with IP details if none of the following are set to prevent wrong combinations
             if (empty($fields['core']['city']['value']) && empty($fields['core']['state']['value']) && empty($fields['core']['country']['value']) && empty($fields['core']['zipcode']['value'])) {
+                if ($this->coreParametersHelper->getParameter('anonymize_ip') && $this->ipLookupHelper->getRealIp()) {
```

Comment:
What about this ? No fatal and not erased if ipLookupDetails is empty.
```suggestion
                if ($this->coreParametersHelper->getParameter('anonymize_ip') && $this->ipLookupHelper->getRealIp()) {
                    $ipLookupDetails = $this->ipLookupHelper->getIpDetails($this->ipLookupHelper->getRealIp());
                    
                    if (!empty($ipLookupDetails)) {
                        $details = $ipLookupDetails;
                    }
                }    
```